content/en/docs/concepts/workloads/pods/probes.md

```diff
@@ -0,0 +1,182 @@
+- **`tcpSocket`**: Specifies the target port for the TCP check.
+- Other fields like `initialDelaySeconds`, `periodSeconds`, `timeoutSeconds`, 
+`failureThreshold`, and `successThreshold` work similarly to HTTP probes.
+
+---
+
+### Exec Probe
+
+An exec probe runs a command inside the container and determines health based on the exit code.
+
+Example Configuration
+```yaml
+startupProbe:
+  exec:
+    command:
+      - cat
+      - /tmp/ready
+  initialDelaySeconds: 10
+  periodSeconds: 5
+```
+
+### Behavior
+- An **exit code of `0`** indicates success.
+- A **non-zero exit code** indicates failure.
+
+---
+
+## gRPC Health Checks
+
+Kubernetes can use the gRPC health check protocol for monitoring gRPC services. The gRPC health 
+check response includes different statuses that can be leveraged for configuring liveliness and
+readiness probes.
+
+### ServiceStatus Fields Used
+Kubernetes can interpret the following gRPC health check statuses:
+
+- **`SERVING`**: The service is healthy.
+- **`NOT_SERVING`**: The service is running but should not receive traffic.
+- **`UNKNOWN`**: The service status is uncertain, and it is generally treated as a failure.
+
+### Customizing Probes with gRPC Health Checks
+It is possible to set up Kubernetes probes such that:
+- A `SERVING` response is considered **ready**.
+- A `SERVING` or `NOT_SERVING` response is considered **alive**.
+- An `UNKNOWN` response will fail both **ready** and **alive** checks.
+
+### Example: Using gRPC Health Probe
+```yaml
+livenessProbe:
+  exec:
+    command:
+      - grpc-health-probe
+      - -addr=127.0.0.1:50051
+      - -service=my-service
+  failureThreshold: 3
+
+readinessProbe:
+  exec:
+    command:
+      - grpc-health-probe
+      - -addr=127.0.0.1:50051
+      - -service=my-service
+  successThreshold: 1
+```
+
+This ensures that if the gRPC service returns `NOT_SERVING`, the liveness check will still pass, 
+keeping the container running, but the readiness probe will fail, removing it from traffic 
+routing.
+
+---
+
+## Debugging Probes
+
+Probes may fail due to misconfiguration or application issues. Use the following commands to inspect the configuration and status of probes:
+
+```bash
+kubectl describe pod <pod-name>
+```
+
+### Common Issues
+- **Incorrect Endpoint Configuration:** Ensure the `path` and `port` are correct for HTTP and TCP probes.
+- **Aggressive Timing Settings:** Probes with short `timeoutSeconds` or `failureThreshold` values can lead to unnecessary restarts.
+- **Application Latency:** Increase `initialDelaySeconds` for slow-starting applications.
+
+---
+
+## Best Practices
+
+Choose the Right Probe for the Use Case:
+   - Use `livenessProbe` for detecting and recovering from container crashes or deadlocks.
+   - Use `readinessProbe` to control traffic routing and ensure requests are only routed to ready containers.
+   - Use `startupProbe` for applications that require significant initialization time.
+
+---
+
+## Learn More
```

Comment:
You should use the way that other pages represent this heading in the Markdown source.